tests/integration/test_store_synapse.py

```diff
@@ -125,3 +132,337 @@ def test_process_row_annotations_get_validation(
+                        "schematic - main/Test files and dataset annotations/Test BulkRNAseq w annotation/txt4.txt",
+                    ),
+                    (
+                        "syn63917520",
+                        "schematic - main/Test files and dataset annotations/Test BulkRNAseq w annotation/txt1.txt",
+                    ),
+                    (
+                        "syn63917521",
+                        "schematic - main/Test files and dataset annotations/Test BulkRNAseq w annotation/txt2.txt",
+                    ),
+                    (
+                        "syn63917522",
+                        "schematic - main/Test files and dataset annotations/Test BulkRNAseq w annotation/txt3.txt",
+                    ),
+                ],
+            ),
+            (
+                "syn23643253",
+                "syn63927665",
+                [
+                    (
+                        "syn63927670",
+                        "schematic - main/BulkRNAseq nested files/data/txt4.txt",
+                    ),
+                    (
+                        "syn63927671",
+                        "schematic - main/BulkRNAseq nested files/data/txt1.txt",
+                    ),
+                    (
+                        "syn63927672",
+                        "schematic - main/BulkRNAseq nested files/data/txt2.txt",
+                    ),
+                    (
+                        "syn63927673",
+                        "schematic - main/BulkRNAseq nested files/data/txt3.txt",
+                    ),
+                ],
+            ),
+            (
+                "syn23643253",
+                "syn63987067",
+                [
+                    (
+                        "syn63987071",
+                        "schematic - main/BulkRNAseq and double nested files/dataset/folder 1/data/txt4.txt",
+                    ),
+                    (
+                        "syn63987072",
+                        "schematic - main/BulkRNAseq and double nested files/dataset/folder 1/data/txt1.txt",
+                    ),
+                    (
+                        "syn63987073",
+                        "schematic - main/BulkRNAseq and double nested files/dataset/folder 1/data/txt2.txt",
+                    ),
+                    (
+                        "syn63987074",
+                        "schematic - main/BulkRNAseq and double nested files/dataset/folder 1/data/txt3.txt",
+                    ),
+                ],
+            ),
+        ],
+        ids=[
+            "top level folder",
+            "nested dataset",
+            "contentType:dataset annotation",
+            "nested data files",
+            "doubly nested data files",
+        ],
+    )
+    @pytest.mark.test_this
+    @pytest.mark.parametrize(
+        "filenames",
+        [None, ["txt1.txt", "txt2.txt"]],
+        ids=["no file filtering", "file filtering"],
+    )
+    def test_getFilesInStorageDataset(
```

Comment:
`test_api.TestManifestOperation.test_generate_manifest_file_based_annotations` tests manifest generation through the api. It tests the nested file dataset structure in a way and sends requests to synapse. The `dataset_id` used in the test is not a dataset, but the `schematic - main` project. This contains other datasets and by our definition is not a valid dataset.
So to also have this case covered after the api migration I included this test and also set up valid datasets on synapse with different file structures.